### PR TITLE
IE-127 Address sonar and dependabot issues in lfp pay api

### PR DESCRIPTION
**Pull request by adr1ancho**
…bot/2

IE-127 address sonar issues in lfp-pay-api

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,33 +13,33 @@ In order to run this API locally you will need to install the following:
 1. Build the executable: `make build`
 
 ## Configuration
-Variable                         | Default | Description
```
Comment (adr1ancho) [inline README.md:16]: This change is performed by IDE.

File: handlers/healthcheck_finance.go [hunk 2]
```diff
@@ -27,53 +27,87 @@ func HandleHealthCheckFinanceSystem(w http.ResponseWriter, r *http.Request) {
 	var systemAvailableTime time.Time
 
 	// Check for weekly downtime
-	if cfg.WeeklyMaintenanceStartTime != "" && cfg.WeeklyMaintenanceEndTime != "" {
```
Comment (adr1ancho) [inline handlers/healthcheck_finance.go:30]: Attempt to reduce complexity by grouping multiple logic checks into one Go function

File: interceptors/payable_resource_authentication.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func (payableAuthInterceptor *PayableAuthenticationInterceptor) PayableAuthentic
 
 		// Get identity type from request
 		identityType := authentication.GetAuthorisedIdentityType(r)
-		if !(identityType == authentication.Oauth2IdentityType || identityType == authentication.APIKeyIdentityType) {
+		if isUnauthorizedIdentityType(identityType) {
```
Comment (adr1ancho) [inline interceptors/payable_resource_authentication.go:42]: Attempt to reduce code complexity by group multiple logic checks into one go function

File: README.md (hunk 1 above)
Comment (adr1ancho) [inline README.md:16]: This is a change made by IDE

File: interceptors/payable_resource_authentication.go [hunk 4]
```diff
@@ -66,29 +65,9 @@ func (payableAuthInterceptor *PayableAuthenticationInterceptor) PayableAuthentic
 			}
 		}
 
-		// Get the payable resource from the ID in request
```
Comment (adr1ancho) [inline interceptors/payable_resource_authentication.go:69]: Attempt to reduce Code complexity by grouping simile logic checks into one Go function

File: interceptors/payable_resource_authentication.go [hunk 5]
```diff
@@ -66,29 +65,9 @@ func (payableAuthInterceptor *PayableAuthenticationInterceptor) PayableAuthentic
...
-		payableResource, responseType, err := payableAuthInterceptor.Service.GetPayableResource(r, companyNumber, payableID)
```
Comment (adr1ancho) [inline interceptors/payable_resource_authentication.go:70]: Attempt to reduce Code complexity by grouping simile logic checks into one Go function

File: validators/payable_transactions.go [hunk 6]
```diff
@@ -62,31 +62,9 @@ func TransactionsArePayable(companyNumber string, txs []models.TransactionItem)
 			return nil, ErrTransactionDoesNotExist
 		}
 
-		if val.IsPartPaid() {
```
Comment (adr1ancho) [inline validators/payable_transactions.go:65]: Attempt to reduce code complexity by grouping similar logic checks into one Go function

**Review by mreis-ch — APPROVED** [review]
All looks good

**Pull request merged**